### PR TITLE
Add media-has-caption rule

### PR DESCRIPTION
**Pull request by evcohen**
Satisfies audio-has-caption in #29 but also works for `<video>` as well.

### CODE REVIEW

**Comment by evcohen** [comment]
@jessebeach I'm having trouble with the flow annotations for this one 😞 may need some help understanding how to resolve

**Comment by evcohen** [comment]
@ljharb I was hoping to implement global settings in this major version so that we could map custom components to DOM elements across all rules, so I held off on implementing in this pass.

But you're right - I think its still valuable to provide per-rule configuration options in case devs want to override the global settings (which I can't think of a case where they would for DOM elements). I'll add the schema in this PR because it definitely makes sense to make use of both global and local config APIs

**Comment by ljharb** [comment]
I don't have a use case for a custom captions prop; ours would be the same for consistency with the DOM. I wouldn't object to that being added, but it might be worth waiting til the use case comes up.

It's worth noting that if all of the tag names and prop names are configurable, then this rule should instead be a custom configuration item in a generic rule about ensuring components have certain props - so I think we want to be careful about allowing 100% configurability :-)

**Comment by evcohen** [comment]
> It's worth noting that if all of the tag names and prop names are configurable, then this rule should instead be a custom configuration item in a generic rule about ensuring components have certain props - so I think we want to be careful about allowing 100% configurability :-)

Haha too true 😂

File: __tests__/src/rules/media-has-caption-test.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+/* eslint-env jest */
```
Comment (ljharb) [inline __tests__/src/rules/media-has-caption-test.js]: this should probably go in `__tests__/.eslintrc` so it doesn't have to be repeated in every file?

Comment (evcohen) [inline __tests__/src/rules/media-has-caption-test.js]: We can do that! It's just a part of the create-rule script so it gets added to every file. Can send a separate PR to eliminate

**Comment by evcohen** [comment]
> (it's your repo so your call, but i'd suggest a rebase to excise the merge commit ;-) )

Haha, I usually squash since I'm not very good at keeping a detailed history.
